noria-server/src/controller/migrate/assignment.rs

```diff
@@ -109,6 +110,11 @@ pub fn assign(
                 };
             }
 
+            // replicas are always in their own domain to distribute the load.
+            if let Some(_) = n.replica_index() {
```

Comment:
This is one of the places where language clarification would help. Is the first `Reader` also a replica? If not, then surely it shouldn't have a `replica_index`? But it does, because currently the first `Reader` is also placed in its own domain?